cpg_utils/hail.py

```diff
@@ -187,3 +187,36 @@ def image_path(suffix: str) -> str:
     prefix = os.getenv('CPG_IMAGE_REGISTRY_PREFIX')
     assert prefix
     return f'{prefix}/{suffix}'
+
+
+def gcloud_authenticate_job(
```

Comment:
In preparation for making this reusable across clouds, can we remove the `gcloud` here? Otherwise we'll have to change all call-sites later once we add Azure support.

Reply:
So reduce this to 'authenticate_job', and in future the same named method can work out how to do the auth (e.g. env variable?)

Reply:
Yes, exactly. We'll also have the added complication of having to authenticate gcloud even from Azure (to talk to the sample metadata server), but I hope that won't change the function declaration (right, @illusional?). We probably don't need to be able to override `key_path: str = '/gsa-key/key.json'` either, as that's really built into Hail Batch.

Reply:
I think this is fine for now, I think Hail Batch is assuming the same path for its Azure key, and yeah we'll have some modifications to make when we support calling to GCP from Azure.